app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,81 @@
+require 'twilio-ruby'
+require 'dotenv'
+
+class TweetsController < ApplicationController
+  get '/tweets' do
+    if !Helpers.is_logged_in?(session)
+      redirect to '/login'
+    else
+      @user = User.find_by_id(session[:user_id])
+      @tweets = Tweet.all
+      erb :'/tweets/index'
+    end
+  end
+
+  get '/tweets/new' do
+    if Helpers.is_logged_in?(session)
+      erb :'/tweets/new'
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content].empty?
+      redirect to '/tweets/new'
+    else
+      @tweet = Tweet.create(content: params[:content], user_id: session[:user_id])
+      # if params[:phone_number] != ""
+      #   @client = Twilio::REST::Client.new(ACCOUNT_SID, AUTH_TOKEN)
+      #   number = "#{params[:phone_number]}"
+      #   @client.messages.create(
+      #     from: 'Number goes here',
+      #     to: number,
+      #     body: params[:content] + "\n - sent by #{@tweet.user.username} from Fwitter"
+      #     )
+      # end
+      redirect to "/tweets/#{@tweet.id}"
+    end
+  end
+
+  get '/tweets/:id' do #tweet show page
+    if !Helpers.is_logged_in?(session)
+      redirect to '/login'
+    else
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/show'
+    end
+  end
+
+  get '/tweets/:id/edit' do #tweet edit page
+    if !Helpers.is_logged_in?(session)
+      redirect to '/login'
+    else
+    @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == session[:user_id]
+        erb :'/tweets/edit'
+      else
+        redirect to '/tweets'
+      end
+    end
+  end
+
+  patch '/tweets/:id' do
+    @tweet = Tweet.find_by_id(params[:id])
+
+    if !params[:content].empty?
+      @tweet.update(content: params[:content])
+      redirect to "/tweets/#{@tweet.id}"
+    else
+      redirect to "/tweets/#{@tweet.id}/edit"
+    end
+  end
+
+  delete '/tweets/:id/delete' do #delete tweet action
+    @tweet = Tweet.find_by_id(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Excellent usage of `find_by_id`!